app/src/main/java/com/keylesspalace/tusky/receiver/SendStatusBroadcastReceiver.kt

```diff
@@ -0,0 +1,140 @@
+
+private const val TAG = "SendStatusBR"
+
+class SendStatusBroadcastReceiver : BroadcastReceiver() {
+
+    @Inject
+    lateinit var accountManager: AccountManager
+
+    override fun onReceive(context: Context, intent: Intent) {
+        AndroidInjection.inject(this, context)
+
+        val notificationId = intent.getStringExtra(NotificationHelper.KEY_NOTIFICATION_ID)
+        val senderId = intent.getLongExtra(NotificationHelper.KEY_SENDER_ACCOUNT_ID, -1)
+        val senderIdentifier = intent.getStringExtra(NotificationHelper.KEY_SENDER_ACCOUNT_IDENTIFIER)
+        val senderFullName = intent.getStringExtra(NotificationHelper.KEY_SENDER_ACCOUNT_FULL_NAME)
+        val citedStatusId = intent.getLongExtra(NotificationHelper.KEY_CITED_STATUS_ID, -1)
+        val visibility = intent.getSerializableExtra(NotificationHelper.KEY_VISIBILITY) as Status.Visibility
+        val spoiler = intent.getStringExtra(NotificationHelper.KEY_SPOILER)
+        val mentions = intent.getStringArrayExtra(NotificationHelper.KEY_MENTIONS)
+        val citedText = intent.getStringExtra(NotificationHelper.KEY_CITED_TEXT)
+        val localAuthorId = intent.getStringExtra(NotificationHelper.KEY_CITED_AUTHOR_LOCAL)
+
+        val account = accountManager.getAccountById(senderId)
+
+        val notificationManager = NotificationManagerCompat.from(context)
+
+
+        if (intent.action == NotificationHelper.REPLY_ACTION) {
+
+            val message = getReplyMessage(intent)
+
+            if (account == null) {
+                Log.w(TAG, "Account \"$senderId\" not found in database. Aborting quick reply!")
+
+                val builder = NotificationCompat.Builder(context, NotificationHelper.CHANNEL_MENTION + senderIdentifier)
+                        .setSmallIcon(R.drawable.ic_notify)
+                        .setColor(ContextCompat.getColor(context, (R.color.primary)))
+                        .setGroup(senderFullName)
+                        .setDefaults(0) // So it doesn't ring twice, notify only in Target callback
+
+                builder.setContentTitle(context.getString(R.string.error_generic))
+                builder.setContentText(context.getString(R.string.error_sender_account_gone))
+
+                builder.setSubText(senderFullName)
+                builder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+                builder.setCategory(NotificationCompat.CATEGORY_SOCIAL)
+                builder.setOnlyAlertOnce(true)
+
+                notificationManager.notify(notificationId.toInt(), builder.build())
+            } else {
+                val text = mentions.joinToString(" ", postfix = " ") { "@$it" } + message.toString()
+
+                val sendIntent = SendTootService.sendTootIntent(
+                        context,
+                        text,
+                        spoiler,
+                        visibility,
+                        false,
+                        Arrays.asList(),
+                        Arrays.asList(),
```

Comment:
you can use `emptyList()` here thats more efficient

Reply:
`listOf` would do the same